bzl/macros/pico.bzl

```diff
@@ -28,13 +26,15 @@ def _pico_elf_and_bin(name, binary, linker_script, **kwargs):
         out = name + "_intermediate.bin",
         target_compatible_with = ["@pico-sdk//bazel/constraint:rp2040"],
     )
-    rp2040_binary(
+
+    platform_transition(
         name = name + ".bin",
-        binary = name + "_bin_intermediate",
+        dep = name + "_bin_intermediate",
+        platform = platform,
         **kwargs
     )
 
-def pico_project(name, srcs, deps, linker_script = "//emb/project/bootloader:application_linker_script", **kwargs):
+def pico_project(name, srcs, deps, platform = "//bzl/platforms:rp2040", **kwargs):
```

Comment:
I'd really love a way to say "of course this is `rp2040` platform, but you know the linker script flag that's set? Let's override it" but nothing I've searched for does that. In fact, if I kept the `pico.bzl` rule around, the docs specifically say that the platform flag will _override_ any transitions.